src/ruby/puppetserver-lib/puppet/server/http_client.rb

```diff
@@ -44,17 +39,29 @@ def self.settings
     @settings ||= {}
   end
 
-  def initialize(server, port, options = {})
-    options = OPTION_DEFAULTS.merge(options)
+  def initialize(options = {})
+    # NOTE: Unlike Puppet's HTTP client implementation, the Java HttpAsyncClient
+    # does not support retries in the version we are using, so the `retry_limit`
+    # option will be ignored if passed.
+    #
+    # Similarly, `redirect_limit` is not easily configurable, so we just use the default
+    # of 100. The param will be ignored if passed.
+  end
 
-    @server = server
-    @port = port
-    @use_ssl = options[:use_ssl]
-    @protocol = @use_ssl ? "https" : "http"
+  def get(url, headers: {}, params: {}, options: {}, &block)
+    request_options = create_common_request_options(url, headers, params, options)
+    java_response = self.class.client_get(request_options)
+    ruby_response = Puppet::Server::HttpResponse.new(java_response,
+                                                     URI(request_options.uri.to_s))
```

Comment:
Since [`create_common_request_options` now raises if `url` is not a `URI`](https://github.com/puppetlabs/puppetserver/pull/2419/files#diff-37e461ce55c05f31a9b64b7eadb239a8ae8261574f3236d3928fd20e8e07485aR89) can we pass `url` directly to the `HttpResponse` constructor?

Reply:
Yeah we can. This was useful in the middle to make sure the query params were making it to Java correctly, but I think it's excessive to leave in.

Reply:
Updated.

Reply:
Oh right, we can't because we haven't attached the query params yet... I assume we want those params included in the URL saved in the response?
```
FAIL in (http-escaped-urls-test) (http_client_test.clj:191)
HTTP GET
expected: (= (str url "/a%20b%3Fc?foo=bar") (.runScriptlet sc "$response.url.to_s"))
  actual: (not (= "http://localhost:40935/a%20b%3Fc?foo=bar" "http://localhost:40935/a%20b%3Fc"))
```